dotCMS/src/main/java/com/dotcms/experiments/business/ExperimentsFactoryImpl.java

```diff
@@ -78,6 +88,48 @@ public Optional<Experiment> find(final String id) throws DotDataException {
         return Optional.of(experiment);
     }
 
+    @Override
+    public List<Experiment> list(ExperimentFilter filter) throws DotDataException {
+        final StringBuilder query = new StringBuilder(LIST_EXPERIMENTS_BASE);
+
+        if(filter.pageId().isPresent()) {
+            query.append(PAGE_ID_FILTER);
+        }
+
+        if(filter.name().isPresent()) {
+            query.append(NAME_FILTER);
+        }
+
+        if(filter.statuses().isPresent()) {
+            if(filter.pageId().isPresent() || filter.name().isPresent()) {
+                query.append("INTERSECT ");
+            }
+
+            final String statusFilter = filter.statuses().get().stream().map((status -> STATUS_FILTER) ).collect
+                    (Collectors.joining(" UNION DISTINCT "));
+            query.append(statusFilter);
```

Comment:
I think this is to complicate why not just concat ' STATUS = ? OR'?, ALSO we avoid the UNION in this case

Reply:
More efficient like this. Set of data is small and could have been like you said but it's working already and tests passing